tasks/image-segmentation-yolov8/src/validate.py

```diff
@@ -0,0 +1,213 @@
+    sample: ImageSample,
+    origSeg: np.ndarray,
+    predictedMask: np.ndarray,
+    sampleAcc: float,
+    artifactImagesPath: Path
+) -> None:
+
+    img = Image.open(sample.imagePath)
+
+    fig, axes = plt.subplots(1, 3, figsize = (20, 10))
+
+    axes[0].imshow(img, cmap = "summer")
+    axes[0].set_title(f"Original image, id: {sample.id}")
+    axes[0].axis("off")
+
+    axes[1].imshow(origSeg, cmap = "summer")
+    axes[1].set_title(f"Original segmentation")
+    axes[1].axis("off")
+
+    axes[2].imshow(predictedMask, cmap = "summer")
+    axes[2].set_title(f"Predicted segmentation\nAcc: {sampleAcc:.2f}")
+    axes[2].axis("off")
+
+    plt.savefig(artifactImagesPath / f"{sample.id}.jpeg")
+    plt.close()
+
+
+def iouScoreClass(testMask: np.ndarray, predictedMask: np.ndarray) -> float:
+    intersecrion = np.logical_and(testMask, predictedMask)
+    union = np.logical_or(testMask, predictedMask)
+
+    return float(np.sum(intersecrion) / np.sum(union))
+
+
+def processPrediction(
+    taskRun: TaskRun[ImageDataset],
+    predictData: list[dict],
+    csvSamplesPath: Path
+) -> None:
+
+    artifactImagesPath = folder_manager.createTempFolder("images")
+
+    fieldNamesSamples = ["sample id", "sample name"]
+    fieldNamesSamples.extend(taskRun.dataset.classes.labels)
+    fieldNamesSamples.append("accuracy")
+
+    with csvSamplesPath.open("w") as file:
+        writer = csv.DictWriter(file, fieldnames = fieldNamesSamples)
+        writer.writeheader()
+
+    for sample in taskRun.dataset.samples:
+        img = Image.open(sample.imagePath)
+        origSeg = sample.load().annotation.extractSegmentationMask(taskRun.dataset.classes)
+
+        predictedMask = np.zeros((img.height, img.width))
+
+        csvSampleValues = [str(sample.id), sample.name]
+
+        iouClasses: list[float] = []
+        for classId, clazz in enumerate(taskRun.dataset.classes):
+            predSampleData = [data for data in predictData if data["image_id"] == sample.id and data["category_id"] == classId]
+
+            if len(predSampleData) <= 0:
+                logging.info(f">> [Image Segmentation] The sample \"{sample.name}\" (sample id: {sample.id}) does not have segmentation for the class \"{clazz.label}\"")
+                continue
+
+            predClassSegmentation = predSampleData[0]
+
+            predClassMask = predClassSegmentation["segmentation"]
+            decodedClassMask = maskCoco.decode(predClassMask)
+            predictedMask = np.logical_or(predictedMask, decodedClassMask).astype(int)
+
+            iouClass = iouScoreClass(sample.load().annotation.extractSegmentationMask(ImageDatasetClasses([clazz])), decodedClassMask)
+            iouClasses.append(iouClass)
+            csvSampleValues.append(f"{iouClass * 100:.2f}")
+
+        if len(iouClasses) > 0:
+            sampleAcc = sum(iouClasses) / len(iouClasses) * 100
+            csvSampleValues.append(f"{sampleAcc:.2f}")
+
+            with csvSamplesPath.open("a") as file:
```

Comment:
`open` should only be called once. Opening the same file inside of for-loop and then only appending one value to it is very slow.